### PR TITLE
Choose sharing name modal refactor

### PR DESCRIPTION
**Pull request by Michael-Holmes-SDE**
Uses the Modal component instead of custom-crafted HTML

### CODE REVIEW

**Comment by Michael-Holmes-SDE** [comment]
I messed up pretty bad lol, so I'm going to redo everything and make a new pull request

**Review by SethPoulsen — APPROVED** [review]

**Comment by SethPoulsen** [comment]
The Failing CI is to do tests of student assessments (in `homework.test.ts`), which is entirely unrelated. Any idea why @nwalters512 ?

**Comment by nwalters512** [comment]
@SethPoulsen #10006 should fix the test failures. It's always something...

**Comment by SethPoulsen** [comment]
Thanks for the fix @nwalters512 , this one should be ready to go now.

File: apps/prairielearn/src/pages/instructorCourseAdminSharing/instructorCourseAdminSharing.html.ts [hunk 1]
```diff
@@ -59,71 +61,53 @@ const addCourseToSharingSetPopover = (resLocals, sharing_set) => {
...
-                  </p>
-                  <p>
-                    Your course's sharing name cannot be changed because at least one question has
-                    been shared. Doing so would break the assessments of other courses that have
-                    imported your questions.
-                  </p>
-                </div>
-              `}
-        </div>
+function chooseSharingNameModal(canChooseSharingName: boolean, csrfToken: string) {
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorCourseAdminSharing/instructorCourseAdminSharing.html.ts:64]: Nit, this should use a `PascalCase` name.

Comment (SethPoulsen) [reply apps/prairielearn/src/pages/instructorCourseAdminSharing/instructorCourseAdminSharing.html.ts]: done, thanks

**Pull request merged**